### PR TITLE
feat: new vertical navigation

### PR DESCRIPTION
**Pull request by lightbringer1991**
Creates a vertical navigation with the ability to collapse/expand.

Sample usage (see screenshot for visual)
```jsx
<VerticalNav isAnimated isCollapsed={this.state.isCollapsed} onClick={this.handleCollapse}>
 <VerticalNav.MenuItem
   onClick={this.handleMenuClick('Tab 1')}
   isActive={this.state.activeTab === 'Tab 1'}
   content={({ isCollapsed }) => (
     <MenuContent // just a custom react component
       {...{
         isCollapsed,
         label: 'Product Details 1',
         icon: <div className="product-details-icon" />,
       }}
     />
   )}
 >
   this is a content
 </VerticalNav.MenuItem>
 <VerticalNav.MenuItem
   onClick={this.handleMenuClick('Tab 2')}
   isActive={this.state.activeTab === 'Tab 2'}
   content={({ isCollapsed }) => (
     <MenuContent
       {...{
         isCollapsed,
         label: 'Product Details 2',
         icon: <div className="product-details-icon" />,
       }}
     />
   )}
 >
   this is another content
 </VerticalNav.MenuItem>
</VerticalNav>
```

TODO:
- [x] tests
- [x] propTypes docs

## Motivation and Background Context

Resolves https://github.com/Adslot/adslot-ui/issues/874

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Screenshots (if appropriate):
after

## Check-list:

- [ ] I have read the [Contributing](CONTRIBUTING.md) document.
- [ ] I've thought about and labelled my PR/commit message appropriately.
- [ ] If this PR introduces breaking changes I've described the impact and migration path for existing applications.
- [ ] CI is green (coverage, linting, tests).
- [ ] I have updated the documentation accordingly.
- [ ] I've two LGTMs/Approvals.
- [ ] I've fixed or replied to all my code-review comments.
- [ ] I've manually tested with a buddy.
- [ ] I've squashed my commits into one.

### CODE REVIEW

**Comment by vinteo** [comment]
LGTM so far

File: src/styles/icons/burger.svg [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (vinteo) [inline src/styles/icons/burger.svg]: don't we usually squeeze these into one line

Comment (lightbringer1991) [reply src/styles/icons/burger.svg]: the bundling converts this into base64 string, so it's ok to leave it as is.

File: src/components/adslot-ui/VerticalNavigation/index.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    return { menuList, contentList: _.compact(contentList) };
+  };
+
+  render() {
+    const { className, dts, isAnimated, isCollapsed } = this.props;
+    const contextValue = { isCollapsed };
+    const { menuList, contentList } = this.splitMenuAndContent();
+
+    const componentClasses = classnames('vertical-navigation-component', className);
```
Comment (veegandhi) [inline src/components/adslot-ui/VerticalNavigation/index.jsx]: `aui--vertical...`

File: src/components/adslot-ui/VerticalNavigation/index.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    const menuClasses = classnames([
+      'vertical-navigation-component__menu',
+      {
+        'is-collapsed': isCollapsed,
```
Comment (veegandhi) [inline src/components/adslot-ui/VerticalNavigation/index.jsx]: `aui--vertical-navigation-component-is-collapsed` ?

File: src/components/adslot-ui/VerticalNavigation/index.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    isCollapsed: false,
+    isAnimated: false,
+  };
+
+  splitMenuAndContent = () => {
+    const { children, isCollapsed } = this.props;
+
+    const menuList = [];
+    const contentList = React.Children.map(children, (child, index) => {
+      if (!child.props.content) {
```
Comment (veegandhi) [inline src/components/adslot-ui/VerticalNavigation/index.jsx]: I think you'll need to do this check for the returned value from `content` as well.

File: src/components/adslot-ui/VerticalNavigation/index.jsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      'vertical-navigation-component__menu',
+      {
+        'is-collapsed': isCollapsed,
+        'is-animated': isAnimated,
+      },
+    ]);
+
+    return (
+      <div className={componentClasses} data-test-selector={dts}>
+        <VerticalNavigationContext.Provider value={contextValue}>
```
Comment (veegandhi) [inline src/components/adslot-ui/VerticalNavigation/index.jsx]: Context is redundant here. The value for `isCollapsed` coming from props.

File: src/components/adslot-ui/VerticalNavigation/index.jsx (hunk 4 above)
Comment (veegandhi) [reply src/components/adslot-ui/VerticalNavigation/index.jsx]: Probably worth type checking `component.type` is `MenuItem`

**Comment by adityasabnis** [comment]
Looking at the example in the description.
I am thinking, since `VerticalNav` can only have one active tab at any point in time. should the `activeTab` should be on VerticalNav instead of MenuItem.

Coz when imp what happens if I pass isActive true for multiple MenuItems?

File: src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import VerticalNavigationContext from '../VerticalNavigationContext';
+
+class MenuItem extends React.PureComponent {
+  static propTypes = {
+    isActive: PropTypes.bool,
+    children: PropTypes.node.isRequired,
+    dts: PropTypes.string,
+
+    // [`content`, `onClick`] is used in VerticalNavigation component
+    content: PropTypes.func.isRequired, // eslint-disable-line react/no-unused-prop-types
```
Comment (adityasabnis) [inline src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx]: Sounds like a mismatch in naming.
`content` should be `element` or `node`.

if it is required as a function, should it be something like `contentHandler` or something on this line?

File: src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    onClick: PropTypes.func.isRequired, // eslint-disable-line react/no-unused-prop-types
+  };
+
+  static defaultProps = {
+    isActive: false,
+  };
```
Comment (adityasabnis) [inline src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx]: As mentioned in the PR comment, I feel, each menuItem should not handle `isActive` state. unless there can be multiple active menuLineItems at any given point in time.

File: src/components/adslot-ui/VerticalNavigation/index.jsx (hunk 4 above)
Comment (adityasabnis) [reply src/components/adslot-ui/VerticalNavigation/index.jsx]: I we do type checking. won't we be introducing the same Tabs and Tab scenario which we discussed a few months back.

Unless we have agreed to go that way moving forward

Comment (veegandhi) [reply src/components/adslot-ui/VerticalNavigation/index.jsx]: `Tab` withing `Tabs`. Is a perfectly valid scenario. So all it needs here is a meaningful message if `MenuItem` is not the child type

File: src/components/adslot-ui/VerticalNavigation/index.jsx [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+    const menuList = [];
+    const contentList = React.Children.map(children, (child, index) => {
+      if (!child.props.content) {
+        // eslint-disable-next-line no-console
+        console.warn('Navigation does not render MenuItem that have no content prop.');
+        return null;
+      }
+
+      const classNames = classnames('vertical-navigation-component__menu-item', { 'is-active': child.props.isActive });
```
Comment (veegandhi) [inline src/components/adslot-ui/VerticalNavigation/index.jsx]: Should have a check somewhere her to set only first tab to `isActive` if multiple `isActive` are passed in.

File: src/components/adslot-ui/VerticalNavigation/index.jsx (hunk 5 above)
Comment (lightbringer1991) [reply src/components/adslot-ui/VerticalNavigation/index.jsx]: @veegandhi I was thinking of not allowing user doing something like this
```jsx
<VerticalNav>
  <VerticalNav.MenuItem isCollapsed />
  <VerticalNav.MenuItem />
  <VerticalNav.MenuItem isCollapsed />
</VerticalNav>

// ideally, all menu should be collapsed/expanded altogether
<VerticalNav isCollapsed>
  <VerticalNav.MenuItem />
  <VerticalNav.MenuItem />
  <VerticalNav.MenuItem />
</VerticalNav>
```

File: src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx (hunk 6 above)
Comment (lightbringer1991) [reply src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx]: this is a renderer for the Menu, happy to change to another name.
@veegandhi what do you think? I have my eyes on `menuRenderer`

Comment (veegandhi) [reply src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx]: @adityasabnis - You're right `content` is required as function. However, the expected response is node. We use functions to expose internal component state `isCollapsed`.`VerticalNav.MenuItem` is providing context, so for this instance just `content` will work.

Comment (adityasabnis) [reply src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx]: ok np

File: src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx (hunk 7 above)
Comment (adityasabnis) [reply src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx]: ^ ?

File: src/styles/icons/details.svg [hunk 9]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (adityasabnis) [inline src/styles/icons/details.svg]: optimize svg using svgo?

File: src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx (hunk 7 above)
Comment (lightbringer1991) [reply src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx]: as per offline discussion with @veegandhi , we want to make the active controlled from the outside, that way we don't need to pass in the tab identifier. Internal logic will restrict the tab to have only 1 tab active at a time.

File: src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class MenuItem extends React.PureComponent {
```
Comment (veegandhi) [inline src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx]: Turn this into a functional component.

Comment (lightbringer1991) [reply src/components/adslot-ui/VerticalNavigation/MenuItem/index.jsx]: it's done

**Pull request merged**